### PR TITLE
API-5193: play 2.8 upgrade

### PR DESCRIPTION
**Pull request by mattclark-zerogravit**

### CODE REVIEW

File: app/uk/gov/hmrc/pushpullnotificationsgateway/connectors/OutboundProxyConnector.scala [hunk 1]
```diff
@@ -104,15 +101,16 @@ class OutboundProxyConnector @Inject()(appConfig: AppConfig,
 object OutboundProxyConnector {
   implicit val callbackValidationResponseFormat: OFormat[CallbackValidationResponse] = Json.format[CallbackValidationResponse]
   private[connectors] case class CallbackValidationResponse(challenge: String)
+  val objLogger = Logger("outboundProxyConnector")
```
Comment (AndySpaven) [inline app/uk/gov/hmrc/pushpullnotificationsgateway/connectors/OutboundProxyConnector.scala]: Any reason not to use logger - the object OutboundProxyConnector can extend AppLogger I seem to recall.

File: app/uk/gov/hmrc/pushpullnotificationsgateway/util/ApplicationLogger.scala [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.pushpullnotificationsgateway.util
+
+import play.api.Logger
+
+trait ApplicationLogger {
+  val logger = Logger("application")
```
Comment (AndySpaven) [inline app/uk/gov/hmrc/pushpullnotificationsgateway/util/ApplicationLogger.scala]: I wonder if we should have made this lazy across the board ??

File: test/uk/gov/hmrc/pushpullnotificationsgateway/connectors/OutboundProxyConnectorSpec.scala [hunk 3]
```diff
@@ -16,34 +16,31 @@
...
-
 import uk.gov.hmrc.pushpullnotificationsgateway.config.AppConfig
 import uk.gov.hmrc.pushpullnotificationsgateway.models._
+import util.HmrcSpec
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future.{failed, successful}
 
-class OutboundProxyConnectorSpec extends WordSpec with Matchers with MockitoSugar with ArgumentMatchersSugar {
+class OutboundProxyConnectorSpec extends HmrcSpec with MockitoSugar with ArgumentMatchersSugar {
```
Comment (AndySpaven) [inline test/uk/gov/hmrc/pushpullnotificationsgateway/connectors/OutboundProxyConnectorSpec.scala]: MS and AMS are not required (HMRCSpec includes them). Appreciate this is not your code.

File: test/uk/gov/hmrc/pushpullnotificationsgateway/connectors/ProxiedHttpClientSpec.scala [hunk 4]
```diff
@@ -19,13 +19,13 @@ package uk.gov.hmrc.pushpullnotificationsgateway.connectors
...
 import org.scalatestplus.play.guice.GuiceOneAppPerSuite
-
 import play.api.inject.guice.GuiceApplicationBuilder
 import play.api.libs.ws.{WSClient, WSRequest}
 import play.api.{Application, Configuration}
 import uk.gov.hmrc.play.audit.http.HttpAuditing
+import util.HmrcSpec
 
-class ProxiedHttpClientSpec extends WordSpec with Matchers with GuiceOneAppPerSuite with OptionValues {
+class ProxiedHttpClientSpec extends HmrcSpec with GuiceOneAppPerSuite with OptionValues {
```
Comment (AndySpaven) [inline test/uk/gov/hmrc/pushpullnotificationsgateway/connectors/ProxiedHttpClientSpec.scala:28]: MS and AMS are not required (HMRCSpec includes them). Appreciate this is not your code.

File: test/uk/gov/hmrc/pushpullnotificationsgateway/controllers/OutboundNotificationControllerSpec.scala [hunk 5]
```diff
@@ -32,14 +27,18 @@ import play.api.libs.json.{JsValue, Json}
...
 import uk.gov.hmrc.pushpullnotificationsgateway.services.CallbackValidator
+import util.HmrcSpec
+
+import scala.concurrent.Future
+import scala.concurrent.Future.{failed, successful}
+import scala.util.{Failure, Success, Try}
 
 class OutboundNotificationControllerSpec
-  extends WordSpec with Matchers with MockitoSugar with ArgumentMatchersSugar with BeforeAndAfterEach with GuiceOneAppPerSuite {
+  extends HmrcSpec with MockitoSugar with ArgumentMatchersSugar with BeforeAndAfterEach with GuiceOneAppPerSuite {
```
Comment (AndySpaven) [inline test/uk/gov/hmrc/pushpullnotificationsgateway/controllers/OutboundNotificationControllerSpec.scala]: MS and AMS are not required (HMRCSpec includes them). Appreciate this is not your code.

File: test/uk/gov/hmrc/pushpullnotificationsgateway/services/CallbackValidatorSpec.scala [hunk 6]
```diff
@@ -16,21 +16,19 @@
...
 import uk.gov.hmrc.pushpullnotificationsgateway.connectors.OutboundProxyConnector
 import uk.gov.hmrc.pushpullnotificationsgateway.models.{CallbackValidation, CallbackValidationResult}
+import util.HmrcSpec
+
+import java.util.UUID.randomUUID
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future.{failed, successful}
 
-class CallbackValidatorSpec extends WordSpec with Matchers with MockitoSugar with ArgumentMatchersSugar {
+class CallbackValidatorSpec extends HmrcSpec with MockitoSugar with ArgumentMatchersSugar {
```
Comment (AndySpaven) [inline test/uk/gov/hmrc/pushpullnotificationsgateway/services/CallbackValidatorSpec.scala]: MS and AMS are not required (HMRCSpec includes them). Appreciate this is not your code.

File: test/uk/gov/hmrc/pushpullnotificationsgateway/services/ChallengeGeneratorSpec.scala [hunk 7]
```diff
@@ -17,9 +17,9 @@
 package uk.gov.hmrc.pushpullnotificationsgateway.services
 
 import org.mockito.{ArgumentMatchersSugar, MockitoSugar}
-import org.scalatest.{Matchers, WordSpec}
+import util.HmrcSpec
 
-class ChallengeGeneratorSpec extends WordSpec with Matchers with MockitoSugar with ArgumentMatchersSugar {
+class ChallengeGeneratorSpec extends HmrcSpec with MockitoSugar with ArgumentMatchersSugar {
```
Comment (AndySpaven) [inline test/uk/gov/hmrc/pushpullnotificationsgateway/services/ChallengeGeneratorSpec.scala]: MS and AMS are not required (HMRCSpec includes them). Appreciate this is not your code.

Comment (mattclark-zerogravit) [reply test/uk/gov/hmrc/pushpullnotificationsgateway/services/ChallengeGeneratorSpec.scala]: Done

File: test/uk/gov/hmrc/pushpullnotificationsgateway/services/CallbackValidatorSpec.scala (hunk 6 above)
Comment (mattclark-zerogravit) [reply test/uk/gov/hmrc/pushpullnotificationsgateway/services/CallbackValidatorSpec.scala]: Done

File: test/uk/gov/hmrc/pushpullnotificationsgateway/controllers/OutboundNotificationControllerSpec.scala (hunk 5 above)
Comment (mattclark-zerogravit) [reply test/uk/gov/hmrc/pushpullnotificationsgateway/controllers/OutboundNotificationControllerSpec.scala]: Done

File: test/uk/gov/hmrc/pushpullnotificationsgateway/connectors/ProxiedHttpClientSpec.scala (hunk 4 above)
Comment (mattclark-zerogravit) [reply test/uk/gov/hmrc/pushpullnotificationsgateway/connectors/ProxiedHttpClientSpec.scala:28]: Done

File: test/uk/gov/hmrc/pushpullnotificationsgateway/connectors/OutboundProxyConnectorSpec.scala (hunk 3 above)
Comment (mattclark-zerogravit) [reply test/uk/gov/hmrc/pushpullnotificationsgateway/connectors/OutboundProxyConnectorSpec.scala]: Done

File: app/uk/gov/hmrc/pushpullnotificationsgateway/util/ApplicationLogger.scala (hunk 2 above)
Comment (mattclark-zerogravit) [reply app/uk/gov/hmrc/pushpullnotificationsgateway/util/ApplicationLogger.scala]: Done

File: app/uk/gov/hmrc/pushpullnotificationsgateway/connectors/OutboundProxyConnector.scala (hunk 1 above)
Comment (mattclark-zerogravit) [reply app/uk/gov/hmrc/pushpullnotificationsgateway/connectors/OutboundProxyConnector.scala]: Have changed this

File: it/uk/gov/hmrc/pushpullnotificationsgateway/connectors/ConnectorSpec.scala [hunk 8]
```diff
@@ -14,12 +14,13 @@
...
 import util._
 
 abstract class ConnectorSpec
-    extends AsyncHmrcSpec 
-    with NoMetricsGuiceOneAppPerSuite 
+    extends AsyncHmrcSpec
+    with NoMetricsGuiceOneAppPerSuite
     with WireMockSugar
-    with WireMockExtensions {}
+    with WireMockExtensions  with MockitoSugar with ArgumentMatchersSugar {}
```
Comment (AndySpaven) [inline it/uk/gov/hmrc/pushpullnotificationsgateway/connectors/ConnectorSpec.scala]: These shouldn't be needed with AsyncHmrcSpec

File: test-common/util/HmrcSpec.scala [hunk 9]
```diff
@@ -16,13 +16,13 @@
...
-import org.scalatest.{Matchers, OptionValues, WordSpec}
+import org.scalatest.OptionValues
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
 import org.scalatestplus.play.WsScalaTestClient
-
 import play.api.test.{DefaultAwaitTimeout, FutureAwaits}
 
-abstract class HmrcSpec extends WordSpec with Matchers with OptionValues with WsScalaTestClient with MockitoSugar with ArgumentMatchersSugar
+abstract class HmrcSpec extends AnyWordSpec with Matchers with OptionValues with WsScalaTestClient
```
Comment (AndySpaven) [inline test-common/util/HmrcSpec.scala]: Ah, the MS and AMS were needed here.  Just not on individual tests...

File: it/uk/gov/hmrc/pushpullnotificationsgateway/connectors/ConnectorSpec.scala (hunk 8 above)
Comment (mattclark-zerogravit) [reply it/uk/gov/hmrc/pushpullnotificationsgateway/connectors/ConnectorSpec.scala]: OK will fix

File: test-common/util/HmrcSpec.scala (hunk 9 above)
Comment (mattclark-zerogravit) [reply test-common/util/HmrcSpec.scala]: Yep I get what you mean now! Will fix

**Review by AndySpaven — APPROVED** [review]

**Pull request merged**